Tests/Unit/Routing/ElasticsearchRouteProviderTest.php

```diff
@@ -0,0 +1,140 @@
+                    ],
+                    [
+                        '_id' => '3',
+                        'title' => 'Foo',
+                        'url' => '/foo',
+                    ],
+                ],
+            ],
+        ];
+    }
+    /**
+     * Tests getters and setters
+     */
+    public function testGettersAndSetters()
+    {
+        $routeMap = ['1', '2', '3'];
+        $collector = $this->getMockBuilder('ONGR\ElasticsearchBundle\Mapping\MetadataCollector')
+            ->disableOriginalConstructor()
+            ->getMock();
+        $manager = $this->getMockBuilder('ONGR\ElasticsearchBundle\Service\Manager')
+            ->disableOriginalConstructor()
+            ->getMock();
+        $routeProvider = new ElasticsearchRouteProvider($collector, $routeMap);
+
+        $routeProvider->setManager($manager);
+
+        $retrievedManager = $routeProvider->getManager();
+        $retrievedMap = $routeProvider->getRouteMap();
+
+        $this->assertEquals($retrievedMap, $routeMap);
+        $this->assertEquals($retrievedManager, $manager);
+    }
+
+    /**
+     * Tests the error when trying to get a route by name
+     *
+     * @expectedException \Symfony\Component\Routing\Exception\RouteNotFoundException
+     */
+    public function testGetRouteByName()
+    {
+        $collector = $this->getMockBuilder('ONGR\ElasticsearchBundle\Mapping\MetadataCollector')
+            ->disableOriginalConstructor()
+            ->getMock();
+        $routeProvider = new ElasticsearchRouteProvider($collector);
+        $routeProvider->getRouteByName('any route');
+    }
+
+    /**
+     * Tests if getRoutes returns an empty collection
+     */
+    public function testGetRoutesByName()
+    {
+        $collector = $this->getMockBuilder('ONGR\ElasticsearchBundle\Mapping\MetadataCollector')
+            ->disableOriginalConstructor()
+            ->getMock();
+        $routeProvider = new ElasticsearchRouteProvider($collector);
+        $names = ['any', 'array', 'of', 'names'];
+
+        $this->assertEquals(
+            new RouteCollection(),
+            $routeProvider->getRoutesByNames($names)
+        );
+    }
+
+    /**
+     * Tests getRouteCollectionForRequest exception when routeMap is not loaded
+     *
+     * @expectedException \Symfony\Component\Routing\Exception\RouteNotFoundException
+     */
+    public function testNoRouteMapDefined()
+    {
+        $collector = $this->getMockBuilder('ONGR\ElasticsearchBundle\Mapping\MetadataCollector')
+            ->disableOriginalConstructor()
+            ->getMock();
+        $provider = new ElasticsearchRouteProvider($collector, []);
+        $provider->setManager($this->getManager());
+        $request = Request::create(
+            '/acme',
+            'GET'
+        );
+        $provider->getRouteCollectionForRequest($request);
+
+    }
+
+    /**
+     * Tests getRouteCollectionForRequest exception manager is undefined
+     *
+     * @expectedException \Exception
```

Comment:
Test is ok, but the `Exception` in code is not correct. Maybe use Logical exception instead.. Or try other some other which is more descrittive.